module-gui/gui/widgets/text/core/cursors/TextCursor.hpp

```diff
@@ -49,6 +49,14 @@ namespace gui
         /// removeChar)
         virtual Move moveCursor(NavigationDirection direction);
         virtual Move moveCursor(NavigationDirection direction, unsigned int n);
+        virtual bool handleNextLine()
+        {
+            return false;
+        };
+        virtual bool handlePreviousLine()
+        {
+            return false;
+        };
```

Comment:
1. move to cpp 
2. consider making this class pure virtual

Reply:
1. Sure. 
2. Hmm but text cursor can exists alone so I see no point? TextLine can be virtual.

Reply:
to be discussed I think :)